client-side/src/components/date/GenericDate.jsx

```diff
@@ -0,0 +1,83 @@
+
+import React, { useState } from 'react';
+import axios from 'axios';
+import Select from '../../stories/Select/Select.jsx';
+import GenericButton from '../../stories/Button/GenericButton.jsx';
+import PropTypes from 'prop-types';
+import formatDate from './formatDate.js';
+
+const DateFormatter = ({ currentUser, date }) => {
+  const { emailFrequency, sendNotificationTime, soundVoice, _id } = currentUser.preferences;
+  const { formatedDate } = currentUser;
+  const [dateFormat, setDateFormat] = useState(formatedDate || 'yyyy-MM-dd');
+
+  const url = process.env.REACT_APP_BASE_URL;
+
+  const dateFormats = [
+    { value: 'yyyy-MM-dd', label: 'yyyy-MM-dd' },
+    { value: 'dd/MM/yyyy', label: 'dd/MM/yyyy' },
+    { value: 'do MMMM yyyy', label: 'do MMMM yyyy' },
+    { value: 'EEEE, MMMM do, yyyy', label: 'EEEE, MMMM do, yyyy' },
+    { value: 'yyyy-MM-dd HH:mm:ss', label: 'yyyy-MM-dd HH:mm:ss' },
+  ];
+
+  const sendFormatDate = async () => {
+    try {
+      const response = await axios.put(`${url}/users/669f86c302f253ec5a6f5162`);
+    } catch (error) {
+      console.error('Error sending format date:', error);
+    }
+  };
+
+  return (
+    <div className="dateWrapper">
+      <Select
+        className='select-data-frequency'
+        options={dateFormats.map(formatOption => ({
+         text: formatOption.label,
+         value: formatOption.value,
+          }))}
+        value={dateFormat}
+        onChange={e => setDateFormat(e.target.value)}
+        title='Select date format'
+        size='large'
+        widthOfSelect='210px'
+      />
+      <GenericButton size='medium' label='send format date' onClick={sendFormatDate} />
+      <div>{formatDate(dateFormat, date)}</div>
```

Comment:
you dont have to save the date separately, the save is for the whole settings component
as well, remove the div